app/models/user_api_key.rb

```diff
@@ -31,6 +31,24 @@ def key_available?
     @key.present?
   end
 
+  def ensure_allowed!(env)
+    raise Discourse::InvalidAccess.new if !allow?(env)
+  end
+
+  def update_last_used!(client_id)
```

Comment:
ActiveRecord uses "bang" methods to raise an error if the record fails to save. Should we do the same here?